detectors/aws/xray/EcsDetector.php

```diff
@@ -0,0 +1,120 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Detectors\Aws\Xray;
+
+use OpenTelemetry\Sdk\Resource\ResourceConstants;
+use OpenTelemetry\Sdk\Resource\ResourceInfo;
+use OpenTelemetry\Sdk\Trace\Attributes;
+
+/**
+ * The AwsEcsDetector can be used to detect if a process is running in AWS
+ * ECS and return a {@link Resource} populated with data about the ECS
+ * plugins of AWS ˜X-Ray. Returns an empty Resource if detection fails.
+ */
+class EcsDetector
+{
+    private const ECS_METADATA_KEY_V4 = 'ECS_CONTAINER_METADATA_URI_V4';
+    private const ECS_METADATA_KEY_V3 = 'ECS_CONTAINER_METADATA_URI';
+
+    private const CONTAINER_ID_LENGTH = 64;
+    private const EMPTY_STRING = '';
+
+    public function __construct(GetProcessData $processData)
+    {
+        $this->processData = $processData;
+    }
+    
+    /**
+     * If running on ECS, runs getContainerId(), getClusterName(), and returns true.
+     * If not running on ECS, returns false.
+     */
+    public function detect(): ResourceInfo
+    {
+        // Check if running on ECS by looking for below environment variables
+        if (!getenv(self::ECS_METADATA_KEY_V4) && !getenv(self::ECS_METADATA_KEY_V3)) {
+            // TODO: add 'Process is not running on ECS' when logs are added
+            return ResourceInfo::emptyResource();
+        }
+
+        $hostName = $this->processData->getHostname();
+        $containerId = self::getContainerId();
+
+        return !$hostName && !$containerId
+            ? ResourceInfo::emptyResource()
+            : ResourceInfo::create(new Attributes([
+                ResourceConstants::CONTAINER_NAME => $hostName,
+                'CONTAINER_ID' => $containerId,
+            ])); //TODO: add ResourceConstants::CONTAINER_ID => $containerId; to main rep0
+    }
+
+    /**
+     * Returns the docker ID of the container found
+     * in its CGroup file.
+     */
+    private function getContainerId()
+    {
+        try {
+            $cgroupData = $this->processData->getCgroupData();
+
+            if (!$cgroupData) {
+                return null;
+            }
+
+            foreach ($cgroupData as $str) {
+                if (strlen($str) > self::CONTAINER_ID_LENGTH) {
+                    return substr($str, strlen($str) - self::CONTAINER_ID_LENGTH);
+                }
+            }
+        } catch (Exception $e) {
+
+            //TODO: add 'Failed to read container ID' when logging is added
+            return null;
+        }
+
+        return null;
+    }
+}
+
+/**
+ * Separated from above class to be able to test using
+ * mock values through unit tests
+ */
+class GetProcessData
```

Comment:
```suggestion
class ProcessDataProvider
```

Class names are generally nouns instead of verbs.